modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C

```diff
@@ -30,28 +33,66 @@ INSFVMomentumDiffusion::validParams()
                              "Switch that can select face interpolation method for the viscosity.");
 
   params.set<unsigned short>("ghost_layers") = 2;
+  params.addParam<bool>(
+      "complete_expansion",
+      false,
+      "Boolean parameter to use complete momentum expansion is the diffusion term.");
+  params.addCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
   return params;
 }
 
 INSFVMomentumDiffusion::INSFVMomentumDiffusion(const InputParameters & params)
   : INSFVFluxKernel(params),
     _mu(getFunctor<ADReal>(NS::mu)),
-    _mu_interp_method(Moose::FV::selectInterpolationMethod(getParam<MooseEnum>("mu_interp_method")))
+    _mu_interp_method(
+        Moose::FV::selectInterpolationMethod(getParam<MooseEnum>("mu_interp_method"))),
+    _u_var(params.isParamValid("u")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))
+               : nullptr),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
```

Comment:
AFAICT these dynamic casts are not needed for functionality. Storing as pointers to ordinary functors should suffice

Reply:
Thanks. Changed to general functors